ord_schema/message_helpers.py

```diff
@@ -520,6 +520,88 @@ def set_compound_smiles(compound, value):
                                    value)
 
 
+def is_transition_metal(atom):
+    """Determines if an atom is a transition metal.
+    
+    Args:
+        atom: The atom in question. Should be of type rdkit.Chem.rdchem.Atom
+        
+    Returns:
+        Boolean for whether the atom is a transition metal.
+    """
+    n = atom.GetAtomicNum()
+    return (n>=22 and n<=29) or (n>=40 and n<=47) or (n>=72 and n<=79)
+
+
+def has_transition_metal(mol):
+    """Determines if a molecule contains a transition metal.
+    
+    Args:
+        mol: The molecule in question. Should be of type rdkit.Chem.rdchem.Mol
+        
+    Returns:
+        Boolean for whether the molecule has a transition metal.
+    """
+    for atom in mol.GetAtoms():
+        if is_transition_metal(atom):
+            return True
+    return False
+
+
+def set_dative_bonds(mol, from_atoms=(7,15)):
+    """Convert metal-ligand bonds to dative. Replaces some single bonds between
+    metals and atoms with atomic numbers in fromAtoms with dative bonds.
+    For all atoms except carbon, the replacement is only done if the atom has
+    "too many" bonds. To handle metal-carbene complexes, metal-carbon bonds
+    are converted to dative if the sum of the explicit and implicit valence
+    of the carbon atom does not equal its default valence, 4.
+
+    Args:
+        mol: The molecule to be converted. 
+        fromAtoms: tuple of atomic numbers corresponding to atom types that
+        should have atom-metal bonds converted to dative. Default is N and P
+                            
+    Returns:
+        The modified molecule.
+    """
+    pt = Chem.GetPeriodicTable()
+    edit_mol = Chem.RWMol(mol)
+    edit_mol.UpdatePropertyCache(strict=False)
+    metals = [atom for atom in edit_mol.GetAtoms() 
+              if is_transition_metal(atom)]
+    for metal in metals:
+        for nbr in metal.GetNeighbors():
+            nbr_atom = nbr.GetAtomicNum()
+            # Handles carbon-bound (e.g., NHC-type) ligands
+            # Converts carbon-metal bond to dative if carbon's total valence +
+            # formal charge does not equal 4
+            if nbr_atom in from_atoms and nbr_atom == 6:
+                if nbr.GetTotalValence() + nbr.GetFormalCharge() \
+                != pt.GetDefaultValence(nbr_atom) and \
+                edit_mol.GetBondBetweenAtoms(nbr.GetIdx(), 
```

Comment:
Can you just update the type of the bond instead of deleting it and adding a new one?

Reply:
I believe this can be done with `ReplaceBond`, but requires creating a new `Bond` object and getting the bond index, etc. Can be done, I only worry about the direction (`BondDir`) of the dative bond being correct. Can check with `GetBondDir` or `GetStartAtom`, but ends up being a bit lengthier than what's there now. Is there another reason to use `ReplaceBond` over this?

Reply:
I think remove/add is a fine solution -- the code looks nice

Reply:
Ah I didn't appreciate the control over the direction. SGTM.